docs/TPM2_PCR_MEASUREMENTS.md

```diff
@@ -43,6 +43,33 @@ recognizable. Measurements currently recorded as `EV_IPL` will continue to be
 recorded as `EV_IPL`, for compatibility reasons. However, `EV_IPL` will not be
 used for new, additional measurements.
 
+## NvPCR Measurements
+
+Since the PCR number space is very small, systemd userspace supports additional
+PCRs implemented via TPM2 NV Indexes (here called *NvPCRs*, even though they
+are no less volatile than classic PCRs), using the TPM2_NT_EXTEND type. These
+mostly behave like real PCRs, but we can allocate them relatively freely from
+the NV index handle space.
+
+The NV index range to use for this is configurable at build time, so that
+downstreams have some flexibility to change this if they want. This uses the
+0x01d10200 NV index as base by default. To abstract the actual nvindex number
+away there's a naming concept, so that nvindexes are referenced by name string
+rather than number.
+
+NvPCRs are defined in little JSON snippets in /usr/lib/nvpcr/*.nvpcr, that
+match up index number and name, as well as pick a hash algorithm.
+
+There's one complication: these NV indexes (like any NV indexes) can be deleted
+by anyone with access to the TPM, and then be recreated. This could be used to
+reset the NvPCRs to zero during runtime, which defeats the whole point of
+them. Our way out: we measure a secret as first thing after creation into the
+NvPCRs. (Or actually, we measure a per-NvPCR secret we derive from a system
+secret via an HMAC of the NvPCR name and the NV index handle). This "anchoring"
+secret is stored in /run/ + /var/lib/ + ESP/XBOOTLDR (the latter encrypted as
+credential, locked to the TPM), to make it available at the whole runtime of
+the OS. It's only accessible to privileged processes with access to the TPM.
```

Comment:
This documentation is a great start. Given it is quite important, I think it would be nice to expand the last sentence a bit. Maybe something like the following:

"Due to this, any process with access to the TPM and read access to any of the storage locations of the anchor secret is considered part of the TCB, as they are able to replay the nvpcr with their own content at will, so due care must be employed when designing a system that uses this feature."